File: C#Integration/PokerProtocol/Commands/Lobby/Response/IdentifyResponse.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using EricUtility;
using EricUtility.Networking.Commands;

namespace PokerProtocol.Commands.Lobby.Response
{
    public class IdentifyResponse : AbstractCommandResponse<IdentifyCommand>
    {

        protected override string CommandName
        {
            get { return IdentifyCommand.COMMAND_NAME; }
        }
        public static string COMMAND_NAME = "lobbyIDENTIFY_RESPONSE";
        private readonly bool m_OK;
        public bool OK
        {
            get { return m_OK; }
        } 


        public IdentifyResponse(StringTokenizer argsToken)
            : base(new IdentifyCommand(argsToken))
        {
            m_OK = bool.Parse(argsToken.NextToken());
        }

        public IdentifyResponse(IdentifyCommand command, bool ok)
            : base(command)
        {
            m_OK = ok;
        }

        public override void Encode(StringBuilder sb)
        {
            base.Encode(sb);
            Append(sb, m_OK);
        }
    }
}

File: JAVA/PokerProtocol/src/protocol/commands/lobby/response/IdentifyResponse.java
package protocol.commands.lobby.response;

import java.util.StringTokenizer;

import protocol.commands.Command;
import protocol.commands.ICommand;

public class IdentifyResponse implements ICommand
{
    public static String COMMAND_NAME = "lobbyIDENTIFY_RESPONSE";
    
    private final String m_playerName;
    private final boolean m_isOK;
    
    public IdentifyResponse(StringTokenizer argsToken)
    {
        m_playerName = argsToken.nextToken();
        m_isOK = Boolean.parseBoolean(argsToken.nextToken());
    }
    
    public IdentifyResponse(String name, boolean ok)
    {
        m_playerName = name;
        m_isOK = ok;
    }
    
    @Override
    public String encodeCommand()
    {
        final StringBuilder sb = new StringBuilder();
        sb.append(IdentifyResponse.COMMAND_NAME);
        sb.append(Command.L_DELIMITER);
        sb.append(m_playerName);
        sb.append(Command.L_DELIMITER);
        sb.append(m_isOK);
        sb.append(Command.L_DELIMITER);
        return sb.toString();
    }
    
    public String getPlayerName()
    {
        return m_playerName;
    }
    
    public boolean isOK()
    {
        return m_isOK;
    }
}
